packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -110,47 +110,29 @@ export default {
          * Segments 1 and 2 have one path to the beginning of `MyComponent` and
          * segment 3 has two paths to the beginning of `MyComponent` since we
          * could have either taken the path of segment 1 or segment 2.
-         *
-         * Populates `cyclic` with cyclic segments.
          */
 
-        function countPathsFromStart(segment) {
-          const {cache} = countPathsFromStart;
-          let paths = cache.get(segment.id);
-
-          // If `paths` is null then we've found a cycle! Add it to `cyclic` and
-          // any other segments which are a part of this cycle.
-          if (paths === null) {
-            if (cyclic.has(segment.id)) {
-              return 0;
-            } else {
-              cyclic.add(segment.id);
-              for (const prevSegment of segment.prevSegments) {
-                countPathsFromStart(prevSegment);
-              }
-              return 0;
-            }
+        function countPathsFromStart(segment, visited = new Set()) {
+          if (codePath.thrownSegments.includes(segment)) {
```

Comment:
Is it okay `includes method` without polyfill?
https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
I guess so, since the linter always runs in a node environmemt.

Reply:
Also, IIRC, thrownSegments isn’t an array here, it’s just array-like.

Reply:
Assuming this is an eslint@5 plugin, then yes as it's supported since node 6.

Reply:
Since it is a direct extract from the original code, this shouldn't really be an issue.

Reply:
Okay, It looks like good.